### PR TITLE
Documentation for the new DX

### LINKED ISSUE #179: docs: Subscriptions - add instructions for listening to row level changes
# Improve documentation

## Link 

https://supabase.io/docs/library/subscribe

## Describe the problem

There are no instructions for listening to row-level updates/inserts/deletes

## Describe the improvement

We need to add the details listed in the readme of the realtime repo: https://github.com/supabase/realtime

```
// Listen to all changes to user ID 99
var allChanges = this.socket.channel('realtime:public:users:id.eq.99')
  .join()
  .on('*', payload => { console.log('Update received!', payload) })
```

## Additional context

I'd have to dig into `supabase-js`, but I suspect it will be something like this:

```js
const mySubscription = supabase
  .from('countries:id.eq.99') // <-- see this
  .on('UPDATE', payload => { console.log('Change received!', payload) })
  .subscribe()
```

### PR DESCRIPTION
**Pull request by kiwicopple**
## What kind of change does this PR introduce?

Updates the docs with the new developer experience: https://deploy-preview-253--supabase.netlify.app/docs

- Documents the new Auth using `gotrue-js`
- Fixes #179 

- [ ] Add OAuth links https://github.com/supabase/supabase/issues/241#issuecomment-705632335 
- [ ] Add docs on Policies

## Additional context

Related PR on `supabase-js` https://github.com/supabase/supabase-js/pull/50

### CODE REVIEW

**Comment by kiwicopple** [comment]
I've moved over the REST calls to the new response format, and in the process I decided to remove the "import" snippets from all the code examples. It think it makes the docs easier to read and focuses the Developer on the specific piece of code they will be using

Change on the left, original on the right.

image

File: web/docs/library/examples/stored.js [hunk 1]
```diff
@@ -5,278 +5,153 @@ import {importJs, errorJs} from './common'
...
-const echoCities = async () => {
-  try {
-    let cities = await supabase
-      .rpc('echo_all_cities')
-      .filter({'name', 'eq', 'Paris'})
-    return cities
-  ${errorJs}
+const { data, error } = await supabase
+  .rpc('echo_all_cities')
+  .filter({'name', 'eq', 'Paris'})
```
Comment (duncanhealy) [inline web/docs/library/examples/stored.js:30]:
```suggestion
  .filter('name', 'eq', 'Paris')
```

File: web/docs/library/examples/delete.js [hunk 2]
```diff
@@ -5,277 +5,162 @@ import {importJs, errorJs} from './common'
...
-    let cities = await supabase
-      .from('cities')
-      .delete()
-      .filter({'name', 'eq', 'Paris'})
-    return cities
-  ${errorJs}
+const { data, error } = await supabase
+  .from('cities')
+  .delete()
+  .filter({'name', 'eq', 'Paris'})
```
Comment (duncanhealy) [inline web/docs/library/examples/delete.js:18]:
```suggestion
  .filter('name', 'eq', 'Paris')
```

**Review by duncanhealy — COMMENTED** [review]
fix filter ({})

**Pull request merged**